x-pack/test/functional/page_objects/lens_page.ts

```diff
@@ -205,5 +220,61 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
       await PageObjects.header.waitUntilLoadingHasFinished();
       await testSubjects.missingOrFail('lnsApp_saveAndReturnButton');
     },
+    /**
+     * Gets label of dimension trigger in dimension panel
+     *
+     * @param dimension - the selector of the dimension
+     */
+    async getDimensionTriggerText(dimension: string, index = 0) {
+      const dimensionElements = await testSubjects.findAll(dimension);
+      const trigger = await testSubjects.findDescendant(
+        'lns-dimensionTrigger',
+        dimensionElements[index]
+      );
+      const text = await trigger.getVisibleText();
+      return text;
+    },
+
+    /**
+     * Gets text of the specified datatable header cell
+     *
+     * @param index - index of th element in datatable
```

Comment:
We don't typically use the position of the element to make assertions, could this be done in a less-brittle way?